src/blocks/sound.rs

```diff
@@ -228,23 +240,46 @@ impl Block for Sound {
     }
 
     fn click(&mut self, e: &I3BarEvent) -> Result<()> {
+
+
         if let Some(ref name) = e.name {
+            let mut command = "".to_string();
```

Comment:
This logic could probably all be moved into the match arm `MouseButton::Left => { ... }` below.